source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+for the BSON type corresponding to the type of the data structure. For example,
+a Python ``datetime`` object must be represented the same as a BSON datetime
+type.
+
+Extended JSON Format
+--------------------
+
+The extended JSON grammar extends the JSON grammar as defined in `section 2`_ of
+the `JSON specification`_ by augmenting the possible JSON values as defined in
+`Section 3`_. An extended JSON value MUST conform to one of the formats given in
+the second column of the following table:
+
+.. _section 2: https://tools.ietf.org/html/rfc7159#section-2
+.. _section 3: https://tools.ietf.org/html/rfc7159#section-3
+
+.. _Canonical Format Table:
+
+Canonical Format Table
+----------------------
+
+Notes
+.....
+
+* Keys within extended JSON values SHOULD be emitted in the order described. All
+  keys MUST be emitted, and any other elements MUST NOT be emitted.
+* Types in *italics* are defined elsewhere in the table or in the `JSON
+  specification`_.
+* Types in <angle brackets> detail the contents of a value.
+
++--------------------+----------------------------------------------------------+
+|**BSON 1.1 Type or  |**Extended JSON Format**                                  |
+|Convention**        |                                                          |
++--------------------+----------------------------------------------------------+
+|ObjectId            |{"$oid": <ObjectId bytes as 23-character, big-endian *hex |
+|                    |string* [#]_>                                             |
++--------------------+----------------------------------------------------------+
+|Symbol              |{"$symbol": *string*}                                     |
++--------------------+----------------------------------------------------------+
+|String              |*string*                                                  |
++--------------------+----------------------------------------------------------+
+|Int32               |{"$numberInt": <32-bit signed integer as a *string*>      |
++--------------------+----------------------------------------------------------+
+|Int64               |{"$numberLong": <64-bit signed integer as a *string*>     |
++--------------------+----------------------------------------------------------+
+|Double              |{"$numberDouble": <64-bit binary floating point as a      |
+|                    |*string*>                                                 |
++--------------------+----------------------------------------------------------+
+|Decimal128          |{"$numberDecimal": <decimal as a *string*>} [#]_          |
++--------------------+----------------------------------------------------------+
+|Binary              |{"$binary": <base64-encoded (with padding as ``=``)       |
+|                    |payload as a *string*, "$type": <BSON binary type as a    |
+|                    |one- or two-character *hex string*}                       |
```

Comment:
Missing symbol fixed. The spec already prefers lowercase hex strings (see note at the bottom of the table). This spec doesn't prevent parsers from parsing uppercase hex strings.